tendrl/node_agent/manager/pull_hardware_inventory.py

```diff
@@ -122,6 +126,95 @@ def getTendrlContext():
     return tendrl_context
 
 
+def get_node_disks():
+    '''returns structure
+
+    {[{"device_name":   "devicename",
+       "fs_type":     "fstype",
+       "fs_uuid":     "uuid",
+       "model":      "model",
+       "mount_point": ["mountpoint", ...],
+       "name":       "name",
+       "parent":     "parentdevicename",
+       "size":       uint64,
+       "disk_type":  "type",
+       "used":       boolean,
+       "ssd":        boolean,
+       "vendor":     "string",
+       "disk_id":     "uuid"}, ...], ...}
+    '''
+    rv = []
+    columns = 'NAME,KNAME,FSTYPE,MOUNTPOINT,UUID,PARTUUID,MODEL,SIZE,TYPE,' \
+              'PKNAME,VENDOR'
+    keys = columns.split(',')
+    lsblk = ("lsblk --all --bytes --noheadings --output='%s' --path --raw" %
+             columns)
+    cmd = Command({"_raw_params": lsblk})
+    out, err = cmd.start()
+    if out:
+        devlist = {}
+        dev_info = {}
+        devlist = map(lambda line: dict(zip(keys, line.split(' '))),
+                      out['stdout'].splitlines())
+        parents = set([d['PKNAME'] for d in devlist if 'PKNAME' in d])
+        for d in devlist:
+            in_use = True
+            if d['TYPE'] not in ['disk', 'part']:
+                continue
+            if d['TYPE'] == 'disk':
+                if d['KNAME'] in parents:
+                    # skip it
+                    continue
+                else:
+                    in_use = False
+            elif not d['FSTYPE']:
+                in_use = False
+
+            d.update({'INUSE': in_use})
+            dev_info.update({d['KNAME']: d})
+
+        for disk in dev_info.values():
+            if disk['TYPE'] == 'disk':
+                ssd_state = is_ssd(disk['KNAME'])
+            else:
+                ssd_state = False
+            if disk["SIZE"] == "":
+                continue
+            rv.append({"device_name": disk["KNAME"],
+                       "fs_type": disk["FSTYPE"],
+                       "fs_uuid": disk["UUID"],
+                       "model": disk["MODEL"],
+                       "mount_point": [disk["MOUNTPOINT"]],
+                       "name": disk["NAME"],
+                       "parent": disk["PKNAME"],
+                       "size": long(disk["SIZE"]),
+                       "disk_type": disk["TYPE"],
+                       "used": disk["INUSE"],
+                       "ssd": ssd_state,
+                       "vendor": disk.get("VENDOR", ""),
+                       "disk_id": disk["UUID"]})
+    else:
+        LOG.error(err)
+    return rv
+
+
+def is_ssd(device):
+    temp = 'cat /sys/block/%s/queue/rotational' % device.split("/")[-1]
+    cmd = Command({"_raw_params": temp})
+    out, err = cmd.start()
+    if not out:
+        LOG.error(err)
+        raise Exception("Failed to get cluster statistics")
+    if out == '0':
+        return True
+    if out == '1':
+        return False
+    """Rotational attribute not found for
```

Comment:
Wrongly Copied from : https://github.com/skyrings/skyring/blob/master/backend/salt/python/skyring/saltwrapper.py#L543

Is this supposed to be a code comment?